tests/airgapped/pipelines/kfp-airgapped.ipynb

```diff
@@ -0,0 +1,223 @@
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import kfp\n",
+    "\n",
+    "from typing import Dict, List\n",
+    "\n",
+    "from kfp import dsl\n",
+    "from kfp.dsl import Input, InputPath, Output, OutputPath, Dataset, Model, component"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1c015f39-61f1-4dfa-b564-5cf1d7f47326",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "client = kfp.Client()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8a48f8bb-431e-4653-b2da-13394d37632e",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "@component(base_image='172.17.0.2:5000/pipelines-runner:ckf-1.8')\n",
+    "def preprocess(\n",
+    "    # An input parameter of type string.\n",
+    "    message: str,\n",
+    "    # Use Output[T] to get a metadata-rich handle to the output artifact\n",
+    "    # of type `Dataset`.\n",
+    "    output_dataset_one: Output[Dataset],\n",
+    "    # A locally accessible filepath for another output artifact of type\n",
+    "    # `Dataset`.\n",
+    "    output_dataset_two_path: OutputPath('Dataset'),\n",
+    "    # A locally accessible filepath for an output parameter of type string.\n",
+    "    output_parameter_path: OutputPath(str),\n",
+    "    # A locally accessible filepath for an output parameter of type bool.\n",
+    "    output_bool_parameter_path: OutputPath(bool),\n",
+    "    # A locally accessible filepath for an output parameter of type dict.\n",
+    "    output_dict_parameter_path: OutputPath(Dict[str, int]),\n",
+    "    # A locally accessible filepath for an output parameter of type list.\n",
+    "    output_list_parameter_path: OutputPath(List[str]),\n",
+    "):\n",
+    "    \"\"\"Dummy preprocessing step.\"\"\"\n",
+    "\n",
+    "    # Use Dataset.path to access a local file path for writing.\n",
+    "    # One can also use Dataset.uri to access the actual URI file path.\n",
+    "    with open(output_dataset_one.path, 'w') as f:\n",
+    "        f.write(message)\n",
+    "\n",
+    "    # OutputPath is used to just pass the local file path of the output artifact\n",
+    "    # to the function.\n",
+    "    with open(output_dataset_two_path, 'w') as f:\n",
+    "        f.write(message)\n",
+    "\n",
+    "    with open(output_parameter_path, 'w') as f:\n",
+    "        f.write(message)\n",
+    "\n",
+    "    with open(output_bool_parameter_path, 'w') as f:\n",
+    "        f.write(\n",
+    "            str(True))  # use either `str()` or `json.dumps()` for bool values.\n",
+    "\n",
+    "    import json\n",
+    "    with open(output_dict_parameter_path, 'w') as f:\n",
+    "        f.write(json.dumps({'A': 1, 'B': 2}))\n",
+    "\n",
+    "    with open(output_list_parameter_path, 'w') as f:\n",
+    "        f.write(json.dumps(['a', 'b', 'c']))"
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
Why is  this dummy training not some real one ? When we already install kfp we can also install scikitlearn and train there some model right ? I dont think sb will use airgapped like this.

Reply:
NOw thinking about it ... it might be too much ... we just want to test pipelines ... WDYT?

Reply:
IMO this test is sufficient because it is covering pipeline components execution and artifacts
This notebook is not for users to play with, it's just for us to use in tests, wdyt?